xds/cmd/controller.go

```diff
@@ -93,20 +155,52 @@ func createFilterChainProvider(
 	)
 	informerFactory.Start(ctx.Done())
 
+	podInformer := informerFactory.Core().V1().Pods().Informer()
+	err := podInformer.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
+		if err == io.EOF {
+			// The informer shutdown successfully.
+			return
+		}
+		logger.WithError(err).Warn("Pod Informer encountered an error")
+	})
+	if err != nil {
+		log.WithError(err).Fatal("failed to set error handler on pod informer")
+	}
+
 	return k8sfilterchain.NewProvider(
 		logger,
 		clock.RealClock{},
 		informerFactory.Core().V1().Pods().Lister(),
-		flags.ProxyPollInterval)
+		flags.ProxyPollInterval), createFilterChainProviderHealthCheck(flags.ProxyNamespace, k8sClient)
+}
+
+func createFilterChainProviderHealthCheck(
+	podNamespace string,
+	k8sClient *kubernetes.Clientset,
+) server.HealthCheck {
+	return func(ctx context.Context) error {
+		if _, err := k8sClient.
```

Comment:
I don't think you need to check both. I think it would be sufficient to check just GameServers (which also checks that the CRDs are installed correctly). I don't think there's a scenario where you can get at one not the other. 

To be devil's advocate for myself -- RBAC rules could restrict access, but you will see that in the logs pretty quick. It's not likely be a runtime issue like a K8s control plane going down. It's all one etcd instance behind Pods and GameServers anyway.

Thoughts?

Reply:
yes its mainly for rbac, I'd rather the pod reports itself as unhealthy if its not able to function properly especially during install/startup or if someone messes with policies at runtime. it seems like an easy win here to be able to detect it